miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/errors/MiniAppError.kt

```diff
@@ -0,0 +1,16 @@
+package com.rakuten.tech.mobile.miniapp.errors
+
+import androidx.annotation.Keep
+
+/**
+ *  Contains the components to use custom errors from host app.
+ *  @property type The type of error.
+ *  @property message error message send to min app.
+ */
+@Keep
+data class MiniAppError(
```

Comment:
It seems like `MiniAppError` should be internal as this is exposing the JSON format that we're using. Also, we want them to only provide the supported types of errors to `getAccessToken`, so I think we need at least three classes:

- `MiniAppBridgeErrorModel` - internal data model (same as `MiniAppError` here currently)
- `MiniAppBridgeError` - abstract public class that's used as base class for errors to pass to mini app
- `MiniAppAccessTokenError` - public class that extends `MiniAppBridgeError` and is used as the error type for `getAccessToken`. It can also have pre-defined errors in the `companion object` to construct the specific errors. (i.e. `MiniAppAccessTokenError.audienceNotSupportedError`)

Do you think that makes sense?